pkg/apis/monitoring/v1alpha1/prometheusagent_types.go

```diff
@@ -91,5 +91,9 @@ func (l *PrometheusAgentList) DeepCopyObject() runtime.Object {
 // https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
 // +k8s:openapi-gen=true
 type PrometheusAgentSpec struct {
+	// +kubebuilder:default="StatefulSet"
+	// +kubebuilder:validation:Enum=StatefulSet;DaemonSet
+	Mode string `json:"mode,omitempty"`
```

Comment:
I'd add a few sentences to set the expectations. I'll also drop the default marker to avoid confusing the users (otherwise they'll get `mode: StatefulSet` in their API responses).
(nit) I realize that the feature gate is `PrometheusAgentDaemonset` while the mode value says Daemon**S**et. Not a big deal but the latter is more [correct](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) so we might want to fix the feature gate name.

```suggestion
        // Mode defines how the Prometheus operator deploys the PrometheusAgent pods. 
        // For now this field has no effect.
        //
        // (Alpha) Using this field requires the `PrometheusAgentDaemonset` feature gate to be enabled.
        //
	// +kubebuilder:validation:Enum=StatefulSet;DaemonSet
	// +optional
	Mode *string `json:"mode,omitempty"`
```

Reply:
Thanks. I fixed all these in the last commit (feature gate is not related to this PR, but it's just a typo fix though)